src/domains/domain.h

```diff
@@ -94,6 +94,11 @@ class domaint
 
     virtual ~valuet() {}
 
+    /// Each abstract value needs to implement clone with covariant return type.
+    /// This is needed to properly clone abstract value when only a pointer to
+    /// valuet is available.
+    virtual valuet *clone()=0;
```

Comment:
:no_entry_sign: Why is the return value a pointer? Who is cleaning up the instances?

Reply:
This is because returning covariant types cannot be implemented with `unique_ptr`. I added a comment that the caller is responsible for taking ownership of the cloned object (ideally by enclosing it into a `unique_ptr`).